src/Microsoft.AspNet.Server.Testing/Deployers/ApplicationDeployer.cs

```diff
@@ -99,41 +99,60 @@ protected string DnxCommandName
 
         protected string PopulateChosenRuntimeInformation()
         {
-            // ex: runtimes/dnx-coreclr-win-x64.1.0.0-rc1-15844/bin
-            var currentRuntimeBinPath = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
-            Logger.LogInformation($"Current runtime path is : {currentRuntimeBinPath}");
-
-            var targetRuntimeName = new StringBuilder()
-                .Append("dnx")
-                .Append((DeploymentParameters.RuntimeFlavor == RuntimeFlavor.CoreClr) ? "-coreclr" : "-clr")
-                .Append($"-{OSPrefix}")
-                .Append((DeploymentParameters.RuntimeArchitecture == RuntimeArchitecture.x86) ? "-x86" : "-x64")
-                .ToString();
-
-            string targetRuntimeBinPath;
-            // Ex: When current runtime is Mono and the tests are being run for CoreClr
-            if (currentRuntimeBinPath.Contains("dnx-mono"))
+            if (TestPlatformHelper.IsMac && DeploymentParameters.RuntimeFlavor == RuntimeFlavor.CoreClr)
```

Comment:
Instead of this, we could following the approach what the MonoDeployer is doing (and probably extract that common logic in the application deployer so that even MonoDeployer can use it)

https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.Server.Testing/Deployers/MonoDeployer.cs#L31-L38